.github/workflows/ci.yaml

```
@@ -63,6 +63,61 @@ jobs:
           bash ./ci/run_tests.sh
           bash ./ci/ensure_copyright.sh
 
+  ubuntu-clang14-address-undefined-leak:
+    runs-on: ubuntu-22.04
+    steps:
+
+      - name: Checkout source
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+          ssh-key: ${{ secrets.SSH_KEY }}
+
+      - name: Install clang-14
+        run: |
+          sudo apt-get update && \
+          sudo apt-get install build-essential software-properties-common -y && \
+          sudo add-apt-repository ppa:ubuntu-toolchain-r/test -y && \
+          sudo apt-get update && \
+          sudo apt-get install clang-14 && \
+          gcc -v
+      - name: Run build
+        env:
+          CC: clang-14
+          CXX: clang++-14
+          TESTENV: clang14
+        run: |
+          bash ./ci/run_address_sanitizer_tests.sh
+
+  # Memory sanitizer is not run because it requires building a custom
+  # instrumented libc.
+
+  ubuntu-clang14-thread:
+    runs-on: ubuntu-22.04
+    steps:
+
+      - name: Checkout source
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+          ssh-key: ${{ secrets.SSH_KEY }}
+
+      - name: Install clang-14
+        run: |
+          sudo apt-get update && \
+          sudo apt-get install build-essential software-properties-common -y && \
+          sudo add-apt-repository ppa:ubuntu-toolchain-r/test -y && \
+          sudo apt-get update && \
+          sudo apt-get install clang-14 && \
+          gcc -v
+      - name: Run build
+        env:
+          CC: clang-14
+          CXX: clang++-14
+          TESTENV: clang14
+        run: |
+          bash ./ci/run_thread_sanitizer_tests.sh
```

extreme nit. This and the `address undefined leak` jobs are relatively similar so we can matricize them. Something like:
```suggestion
  ubuntu-clang14-sanitizer-tests:
    runs-on: ubuntu-22.04
    strategy:
        os:
          - { name: "address-undefined-leak", script: "./ci/run_address_sanitizer_tests.sh" }
          - { name: "thread", script: "./ci/run_thread_sanitizer_tests.sh" }
    steps:

      - name: Checkout source
        uses: actions/checkout@v2
        with:
          submodules: recursive
          ssh-key: ${{ secrets.SSH_KEY }}

      - name: Install clang-14
        run: |
          sudo apt-get update
          sudo apt-get install build-essential software-properties-common -y
          sudo add-apt-repository ppa:ubuntu-toolchain-r/test -y
          sudo apt-get update
          sudo apt-get install clang-14
          gcc -v
      - name: Run (${{ matrix.name }} ) Sanitizer Tests
        env:
          CC: clang-14
          CXX: clang++-14
          TESTENV: clang14
        run: |
          bash "${{ matrix.script }}"
```

---

Done; let's see if I copied correctly :)